Sources/Algorithms/FirstNonNil.swift

```diff
@@ -0,0 +1,26 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// firstNonNil(_:)
+//===----------------------------------------------------------------------===//
+
+public extension Sequence {
+    /// Returns the first element in `self` that `transform` maps to a `.some`.
+    func firstNonNil<Result>(_ transform: (Element) throws -> Result?) rethrows -> Result? {
```

Comment:
Formatting: to be consistent with the rest of the project, please wrap to 80 columns. Also, use 2 spaces instead of 4 for indenting.

```suggestion
  func firstNonNil<Result>(
    _ transform: (Element) throws -> Result?
  ) rethrows -> Result? {
```